subsys/net/lib/sockets/sockets.c

```diff
@@ -845,20 +845,41 @@ static inline int z_vrfy_zsock_inet_pton(sa_family_t family,
 #endif
 
 int z_impl_zsock_getsockopt(int sock, int level, int optname,
-			    void *optval, socklen_t *optlen)
+	void* optval, socklen_t* optlen)
 {
 	int ret;
 
 	SYS_PORT_TRACING_OBJ_FUNC_ENTER(socket, getsockopt, sock, level, optname);
 
+	if (level == IPPROTO_IP && optname == IP_MULTICAST_LOOP) {
+		if (*optlen < sizeof(int)) {
+			errno = EINVAL;
+			ret = -1;
+			goto exit;
+		}
+
+		struct net_context* ctx = zsock_get_context_object(sock);
+		if (ctx == NULL) {
+			errno = EBADF;
+			ret = -1;
+			goto exit;
+		}
+
+		*(int*)optval = ctx->multicast_loop;
+		*optlen = sizeof(int);
+		ret = 0;
+		goto exit;
+	}
+exit:
```

Comment:
Place the option setting / getting to `subsys/net/lib/sockets/sockets_inet.c` where the other options are set